b2/console_tool.py

```diff
@@ -2598,43 +2608,227 @@ def run(self, args):
+        self._print_json(file_info)
+        return 0
+
+    def get_execute_kwargs(self, args) -> dict:
+        file_infos = self._parse_file_infos(args.info)
+
+        if SRC_LAST_MODIFIED_MILLIS not in file_infos and os.path.exists(args.localFilePath):
+            try:
+                mtime = os.path.getmtime(args.localFilePath)
+            except OSError:
+                if not points_to_fifo(pathlib.Path(args.localFilePath)):
+                    self._print_stderr(
+                        "WARNING: Unable to determine file modification timestamp. "
+                        f"{SRC_LAST_MODIFIED_MILLIS!r} file info won't be set."
+                    )
+            else:
+                file_infos[SRC_LAST_MODIFIED_MILLIS] = str(int(mtime * 1000))
+
+        return {
+            "bucket": self.api.get_bucket_by_name(args.bucketName),
+            "cache_control": args.cache_control,
+            "content_type": args.contentType,
+            "custom_upload_timestamp": args.custom_upload_timestamp,
+            "encryption": self._get_destination_sse_setting(args),
+            "file_info": file_infos,
+            "file_name": args.b2FileName,
+            "file_retention": self._get_file_retention_setting(args),
+            "legal_hold": self._get_legal_hold_setting(args),
+            "local_file": args.localFilePath,
+            "min_part_size": args.minPartSize,
+            "progress_listener": make_progress_listener(args.localFilePath, args.noProgress),
+            "sha1_sum": args.sha1,
+        }
+
+    def execute_operation(self, **kwargs) -> 'b2sdk.file_version.FileVersion':
+        raise NotImplementedError
+
+    def upload_file_kwargs_to_unbound_upload(self, **kwargs):
+        """
+        Translate upload_file kwargs to unbound_upload equivalents
+        """
+        kwargs["large_file_sha1"] = kwargs.pop("sha1_sum", None)
+        return kwargs
+
+    def get_input_stream(self, filename: str) -> 'str | int | io.BinaryIO':
+        """Get input stream IF filename points to a FIFO or stdin."""
+        if filename == "-":
+            if os.path.exists('-'):
+                self._print_stderr(
+                    "WARNING: Filename `-` won't be supported in the future and will be treated as stdin alias."
+                )
+            else:
+                return sys.stdin.buffer if platform.system() == "Windows" else sys.stdin.fileno()
+        elif points_to_fifo(pathlib.Path(filename)):
+            return filename
+
+        raise self.NotAnInputStream()
+
+    def file_identifier_to_read_stream(self, file_id: 'str | int | BinaryIO', **kwargs) -> BinaryIO:
+        if isinstance(file_id, (str, int)):
+            return open(
+                file_id,
+                mode="rb",
+                closefd=not isinstance(file_id, int),
+                buffering=kwargs.get('buffering', RECOMMENDED_READ_BUF_SIZE),
+            )
+        return file_id
+
+    class NotAnInputStream(Exception):
```

Comment:
I don't think I've seen his pattern in this project before, any reason for not declaring this exception at module level?

Reply:
because the only function rising it is a class method; i.e. if this function is removed, exception should be removed as well

We are talking about ~4k long file, which is why namespacing is especially important.